sdk/mixedreality/azure-mixedreality-authentication/src/main/java/com/azure/mixedreality/authentication/MixedRealityStsClientBuilder.java

```diff
@@ -0,0 +1,402 @@
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.mixedreality.authentication.implementation.MixedRealityStsRestClientImpl;
+import com.azure.mixedreality.authentication.implementation.MixedRealityStsRestClientImplBuilder;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of {@link
+ * MixedRealityStsClient MixedRealityStsClients} and {@link MixedRealityStsAsyncClient MixedRealityStsAsyncClient}, call {@link
+ * #buildClient() buildClient} and {@link #buildAsyncClient() buildAsyncClient} respectively to construct an instance of
+ * the desired client.
+ *
+ * @see MixedRealityStsAsyncClient
+ * @see MixedRealityStsClient
+ */
+@ServiceClientBuilder(serviceClients = {MixedRealityStsClient.class, MixedRealityStsAsyncClient.class})
+public final class MixedRealityStsClientBuilder {
+    private static final String APP_CONFIG_PROPERTIES = "azure-mixedreality-authentication.properties";
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private final List<HttpPipelinePolicy> customPolicies = new ArrayList<HttpPipelinePolicy>();
+    private final ClientLogger logger = new ClientLogger(MixedRealityStsClientBuilder.class);
+
+    private String accountDomain;
+    private String accountId;
+    private MixedRealityStsServiceVersion apiVersion;
+    private ClientOptions clientOptions;
+    private Configuration configuration;
+    private String endpoint;
+    private HttpClient httpClient;
+    private AzureKeyCredential keyCredential;
+    private HttpLogOptions logOptions = new HttpLogOptions();
+    private HttpPipeline pipeline;
+    private RetryPolicy retryPolicy;
+    private TokenCredential tokenCredential;
+
+    /**
+     * Constructs a new builder used to configure and build {@link MixedRealityStsClient MixedRealityStsClients} and
+     * {@link MixedRealityStsAsyncClient MixedRealityStsAsyncClients}.
+     */
+    public MixedRealityStsClientBuilder() {
+    }
+
+    /**
+     * Sets the Mixed Reality service account domain.
+     *
+     * @param accountDomain The Mixed Reality service account domain.
+     * @return The updated {@link MixedRealityStsClientBuilder} object.
+     * @throws IllegalArgumentException If {@code accountDomain} is null or empty.
+     */
+    public MixedRealityStsClientBuilder accountDomain(String accountDomain) {
+        if (CoreUtils.isNullOrEmpty(accountDomain)) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'accountDomain' cannot be null or empty."));
+        }
```

Comment:
Use NPE for null input and IllegalArgumentException for invalid input - https://azure.github.io/azure-sdk/java_introduction.html#java-errors-system-errors

Reply:
So, should I check for null and empty independently? Throw `NullPointerException` if it's null and `IllegalArgumentException` if it is empty?

Reply:
Yes - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java#L232-L241

Reply:
Thanks for the clarification.